util/db/dbutil.go

```diff
@@ -125,24 +127,30 @@ func Retry(fn func() error) (err error) {
 	}
 }
 
-// Atomic executes a piece of code with respect to the database atomically.
-// For transactions where readOnly is false, sync determines whether or not to wait for the result.
-func (db Accessor) Atomic(fn idemFn) (err error) {
-	descr := "w"
-	if db.readOnly {
-		descr = "r"
+// getDecoratedLogger retruns a decorated logger that includes the readonly true/false, caller and extra fields.
+func (db *Accessor) getDecoratedLogger(fn idemFn, extras ...interface{}) logging.Logger {
+	log := logging.Base().With("readonly", db.readOnly)
+	_, file, line, ok := runtime.Caller(2)
+	if ok {
+		log = log.With("caller", fmt.Sprintf("%s:%d", file, line))
+	}
+	log = log.With("callee", runtime.FuncForPC(reflect.ValueOf(fn).Pointer()).Name())
+	if len(extras) > 0 {
```

Comment:
I think you can drop this `len()` check, since the inner `for` loop will be a noop when `len() == 0` anyway.